tensorflow/core/common_runtime/gpu/gpu_device.cc

```diff
@@ -604,7 +604,7 @@ LocalDevice* BaseGPUDeviceFactory::CreateGPUDevice(
       allocated_memory -= min_system_memory;
     }
   } else {
-    allocated_memory *= config_memory_fraction;
```

Comment:
For clarity (so it's clear that this path doesn't use available_memory at all), can you change 598 to be just 

    int64 allocated_memory;
    if (config_memory_fraction == 0) {
      allocated_memory = available_memory;
      ...
    } else {
      allocated_memory = total_memory * config_memory_fraction;
    }